common/cgroup/cgroup.go

```diff
@@ -401,6 +402,59 @@ func RemountCgroupsRO(root string, enabledCgroups map[int][]string, subcgroup st
 	return mountFsRO(sysPath)
 }
 
+// RemountCgroupKnobsRW remounts the needed knobs in the subcgroup for one
+// specified app read-write so the systemd inside stage1 can apply isolators
+// to them.
+func RemountCgroupKnobsRW(enabledCgroups map[int][]string, subcgroup string, serviceName string, enterCmd []string) error {
+	controllers := GetControllerDirs(enabledCgroups)
+
+	// Mount RW knobs we need to make the enabled isolators work
+	for _, c := range controllers {
+		cPath := filepath.Join("/sys/fs/cgroup", c)
+		subcgroupPath := filepath.Join(cPath, subcgroup, "system.slice")
+
+		// Create cgroup directories and mount the files we need over
+		// themselves so they stay read-write
+		appCgroup := filepath.Join(subcgroupPath, serviceName)
+		if err := os.MkdirAll(appCgroup, 0755); err != nil {
+			return err
+		}
+		for _, f := range getControllerRWFiles(c) {
+			cgroupFilePath := filepath.Join(appCgroup, f)
+			// the file may not be there if kernel doesn't support the
+			// feature, skip it in that case
+			if _, err := os.Stat(cgroupFilePath); os.IsNotExist(err) {
+				continue
+			}
+
+			// Go applications cannot be  reassociated  with a new mount
+			// namespace because they are multithreaded. Instead of
+			// syscall.Mount, uses the enter entrypoint.
+			argsMountBind := append(enterCmd, "/bin/mount", "--bind", cgroupFilePath, cgroupFilePath)
```

Comment:
Just curious since I am not familiar with this: Why we need to do two mounts?
(`mount --bind foo bar` and `mount -o remount,bind,rw`)